src/note.rs

```diff
@@ -282,28 +340,78 @@ impl Note {
 
     /// Derives the nullifier for this note.
     pub fn nullifier(&self, fvk: &FullViewingKey) -> Nullifier {
+        let selected_rseed = self.rseed_split_note.unwrap_or(self.rseed);
+
         Nullifier::derive(
             fvk.nk(),
             self.rho.0,
-            self.rseed.psi(&self.rho),
+            selected_rseed.psi(&self.rho),
             self.commitment(),
+            self.rseed_split_note.is_some(),
         )
     }
+
+    /// Create a split note which has the same values than the input note except for
+    /// `rseed_split_note` which is equal to a random seed.
+    pub fn create_split_note(self, rng: &mut impl RngCore) -> Self {
```

Comment:
_From str4d ([r2543575888](https://github.com/zcash/orchard/pull/471#discussion_r2543575888))_

```suggestion
    pub fn create_split_note(self, rng: &mut impl RngCore) -> Self {
        assert!(bool::from(!self.asset().is_native()));
```

Reply:
TODO